src/Compilers/Core/Portable/PEWriter/CustomDebugInfoWriter.cs

```diff
@@ -125,16 +125,27 @@ internal static void SerializeCustomDebugInformation(EditAndContinueMethodDebugI
         {
             if (!debugInfo.LocalSlots.IsDefaultOrEmpty)
             {
-                customDebugInfo.Add(SerializeRecord(CDI.CdiKindEditAndContinueLocalSlotMap, debugInfo.SerializeLocalSlots));
+                customDebugInfo.Add(
+                    SerializeRecord(
+                        CDI.CdiKindEditAndContinueLocalSlotMap, 
+                        debugInfo,
+                        (info, builder) => info.SerializeLocalSlots(builder)));
```

Comment:
Yes. 
If we enclose debugInfo with the delegate, code looks nicer, but we create a lot of delegates here.